compiler/src/lntoamm/Type.ts

```diff
@@ -502,25 +502,23 @@ export class Type {
     const otherTypeAst = fulltypenameAstFromString(otherType.typename)
     let generics = []
     if (typeAst.has('opttypegenerics')) {
-      const genericsAst = typeAst.get('opttypegenerics').get('generics')
-      generics.push(genericsAst.get('fulltypename').t)
-      genericsAst.get('cdr').getAll().forEach(r => {
+      generics.push(typeAst.get('opttypegenerics').get('generics').get('fulltypename').t)
+      typeAst.get('opttypegenerics').get('generics').get('cdr').getAll().forEach(r => {
         generics.push(r.get('fulltypename').t)
       })
```

Comment:
Somehow this code and the similar block below weren't being run *at all* until now, shockingly. It was translated incorrectly back when I switched from ANTLR and never noticed.